src/contract_client/utils.ts

```diff
@@ -77,8 +81,18 @@ export const contractErrorPattern = /Error\(Contract, #(\d+)\)/;
 /**
  * A TypeScript type guard that checks if an object has a `toString` method.
  */
-export function implementsToString(
-  obj: unknown,
-): obj is { toString(): string } {
+export function implementsToString(obj: unknown): obj is { toString(): string } {
   return typeof obj === "object" && obj !== null && "toString" in obj;
```

Comment:
You need the `obj !== null` because [typeof null === "object"](https://stackoverflow.com/questions/18808226/why-is-typeof-null-object) for hysterical reasons

But yeah, probably better to check that `typeof toString === "function"`

Though I think these lines were just changed because of eslint, so I'm not sure it's worth changing the implementation as part of this PR